CMakeLists.txt

```
@@ -61,7 +62,7 @@ file(
 
 add_library(CoScheduler SHARED ${all_files})
 target_include_directories(
-  CoScheduler PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>
+  CoScheduler PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src> ${Boost_INCLUDE_DIRS}
                      $<INSTALL_INTERFACE:include>)
 
 target_link_libraries(CoScheduler PUBLIC TBB::tbb CUDA::cudart)
```

```suggestion
  CoScheduler PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>
                     $<INSTALL_INTERFACE:include>)

target_link_libraries(CoScheduler PUBLIC TBB::tbb CUDA::cudart Boost::headers)
```
Just a small change, after we find boost we could add headers by linking target